core/src/Kokkos_HostSpace.hpp

```diff
@@ -187,7 +191,7 @@ class SharedAllocationRecord<Kokkos::HostSpace, void>
   static RecordBase s_root_record;
 #endif
 
-  const Kokkos::HostSpace m_space;
+  Kokkos::HostSpace m_space;
```

Comment:
Although I agree with that change in general, why is it part of this PR?

Reply:
The compiler complained about const default constructed objects. This appeared better than explicitly brace-initializing.